File: src/main/java/kr/co/lunatalk/domain/image/dto/response/PresignedUrlResponse.java
package kr.co.lunatalk.domain.image.dto.response;

import io.swagger.v3.oas.annotations.media.Schema;

public record PresignedUrlResponse(
	@Schema(description = "이미지 업로드 presigned url")
	String presignedUrl,
	@Schema(description = "이미지 키")
	String imageKey
) {

	public static PresignedUrlResponse of(String presignedUrl, String imageKey) {
		return new PresignedUrlResponse(presignedUrl, imageKey);
	}
}

File: src/main/kotlin/kr/co/lunatalk/domain/image/dto/response/PresignedUrlResponse.kt
package kr.co.lunatalk.domain.image.dto.response

import io.swagger.v3.oas.annotations.media.Schema

data class PresignedUrlResponse(
    @Schema(description = "이미지 업로드 presigned url")
    val presignedUrl: String,

    @Schema(description = "이미지 키")
    val imageKey: String
) {
    companion object {
        fun of(presignedUrl: String, imageKey: String): PresignedUrlResponse {
            return PresignedUrlResponse(presignedUrl, imageKey)
        }
    }
}
